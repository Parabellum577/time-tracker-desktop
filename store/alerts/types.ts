import { ActionType } from 'typesafe-actions'

import * as actions from './actions'
import { IError } from '@services/types'

export type AlertsAction = ActionType<typeof actions>

export interface IAlert {
  alertType: 'success' | 'error' | 'warning' | 'coldstart'
  alertMessage: string | IError
}

export interface IAlertsState {
  alertsArray: IAlert[]
}
