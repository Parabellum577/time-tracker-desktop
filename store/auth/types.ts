import { ActionType } from 'typesafe-actions'
import { DeepReadonly } from 'utility-types'

import * as actions from './actions'
import { IError } from '@types'

export type AuthAction = ActionType<typeof actions>

export type IAuthState = DeepReadonly<{
  readonly authorized: boolean
  readonly error?: IError | null
  readonly loading: boolean
  readonly isLoadingData: boolean
}>
