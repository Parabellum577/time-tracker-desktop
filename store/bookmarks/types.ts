import { IFullIssue } from '@types'

export interface IBookmarks {
  extendedBookmarks: IFullIssue[]
}
