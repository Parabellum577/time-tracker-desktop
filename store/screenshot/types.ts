import { IUserScreenshotSettings } from '@services/types'

export interface IScreenShotStorageState {
  availableStorage: IUserScreenshotSettings
  status: boolean
}
