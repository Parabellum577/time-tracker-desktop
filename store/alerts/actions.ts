import { createAsyncAction } from 'typesafe-actions'
import { IError } from '@types'
import { IAlert } from './types'

export const showAlertMessage = createAsyncAction(
  'notifications/SHOW_ALERT_MESSAGE_REQUEST',
  'notifications/SHOW_ALERT_MESSAGE_SUCCESS',
  'notifications/SHOW_ALERT_MESSAGE_FAILURE',
)<IAlert, IAlert, IError>()

export const clearAlertMessage = createAsyncAction(
  'notifications/CLEAR_ALERT_MESSAGE_REQUEST',
  'notifications/CLEAR_ALERT_MESSAGE_SUCCESS',
  'notifications/CLEAR_ALERT_MESSAGE_FAILURE',
)<void, void, IError>()
