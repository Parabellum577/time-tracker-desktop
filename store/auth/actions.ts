import { createAction, createAsyncAction } from 'typesafe-actions'
import { IError } from '@types'
import { ILoginUserPayload, ILoginOTSPayload } from '@services/api-types'

export const loginUser = createAsyncAction('auth/LOGIN_REQUEST', 'auth/LOGIN_SUCCESS', 'auth/LOGIN_FAILURE')<
  ILoginUserPayload,
  void,
  IError
>()

export const loginUserViaOAuth = createAsyncAction(
  'auth/LOGIN_VIA_OAUTH_REQUEST',
  'auth/LOGIN_VIA_OAUTH_SUCCESS',
  'auth/LOGIN_VIA_OAUTH_FAILURE',
)<void, void, IError>()

export const otsLogin = createAsyncAction('auth/OTS_LOGIN_REQUEST', 'auth/OTS_LOGIN_SUCCESS', 'auth/OTS_LOGIN_FAILURE')<
  ILoginOTSPayload,
  ILoginOTSPayload,
  IError
>()

export const loadingDataAfterLogin = createAction('auth/LOADING_DATA_AFTER_LOGIN', resolve => (isLoading: boolean) =>
  resolve(isLoading),
)

export const signOut = createAction('auth/SIGN_OUT', resolve => () => resolve())
