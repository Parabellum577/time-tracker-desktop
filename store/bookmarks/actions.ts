import { createAsyncAction, createAction } from 'typesafe-actions'
import { IError, IFullIssue } from '@types'

export const getBookmarks = createAsyncAction(
  'bookmarks/GET_BOOKMARKS_REQUEST',
  'bookmarks/GET_BOOKMARKS_SUCCESS',
  'bookmarks/GET_BOOKMARKS_FAILURE',
)<void, IFullIssue[], IError>()

export const handleBookmark = createAsyncAction(
  'bookmarks/HANDLE_BOOKMARK_REQUEST',
  'bookmarks/HANDLE_BOOKMARK_SUCCESS',
  'bookmarks/HANDLE_BOOKMARK_FAILURE',
)<IFullIssue, IFullIssue, IError>()

export const clearBookmarks = createAction('plannings/CLEAR_BOOKMARKS', action => () => action())
