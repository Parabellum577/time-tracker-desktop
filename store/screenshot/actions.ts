import { createAsyncAction, createAction } from 'typesafe-actions'
import { IError, IScreenshotStatus, IUserScreenshotSettings } from '@types'

export const getScreenshotStorage = createAsyncAction(
  'screenshot/GET_SCREENSHOT_STORAGE_REQUEST',
  'screenshot/GET_SCREENSHOT_STORAGE_SUCCESS',
  'screenshot/GET_SCREENSHOT_STORAGE_FAILURE',
)<void, IUserScreenshotSettings, IError>()

export const addScreenshotStorage = createAsyncAction(
  'screenshot/ADD_SCREENSHOT_STORAGE_REQUEST',
  'screenshot/ADD_SCREENSHOT_STORAGE_SUCCESS',
  'screenshot/ADD_SCREENSHOT_STORAGE_FAILURE',
)<void, IUserScreenshotSettings, IError>()

export const deleteScreenshotStorages = createAction(
  'screenshot/DELETE_SCREENSHOT_STORAGE',
  resolve => () => resolve(),
)

export const handleScreenshotStatus = createAsyncAction(
  'screenshot/HANDLE_SCREENSHOT_STATUS_REQUEST',
  'screenshot/HANDLE_SCREENSHOT_STATUS_SUCCESS',
  'screenshot/HANDLE_SCREENSHOT_STATUS_FAILURE',
)<IScreenshotStatus, IScreenshotStatus, IError>()
