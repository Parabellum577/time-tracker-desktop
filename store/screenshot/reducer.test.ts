import { ActionType, getType } from 'typesafe-actions'

import * as actions from './actions'
import { IScreenShotStorageState } from './types'
import screenshots from './reducer'
import { SIMPLE_SCREENSHOT_STORAGE } from '@test-helpers/sample'

describe('[Store] Screenshot Reducer', () => {
  it('[Success] [Snapshot] Should handle getScreenshotStorage', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getScreenshotStorage.success),
      payload: { ...SIMPLE_SCREENSHOT_STORAGE },
    }

    const currentState: IScreenShotStorageState = {
      availableStorage: { ...SIMPLE_SCREENSHOT_STORAGE },
      status: true,
    }

    expect(screenshots(currentState, action)).toMatchSnapshot()
  })

  it('[Failure] Should handle getScreenshotStorage', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.getScreenshotStorage.failure),
      payload: {
        code: 666,
        message: 'SOME_ERROR',
      },
    }

    const currentState: IScreenShotStorageState = {
      availableStorage: null,
      status: true,
    }

    expect(screenshots(currentState, action)).toMatchSnapshot()
  })

  it('[Success] [Snapshot] Should handle addScreenshotStorage', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.addScreenshotStorage.success),
      payload: { ...SIMPLE_SCREENSHOT_STORAGE },
    }

    const currentState: IScreenShotStorageState = {
      availableStorage: { ...SIMPLE_SCREENSHOT_STORAGE },
      status: true,
    }

    expect(screenshots(currentState, action)).toMatchSnapshot()
  })

  it('[Success] [Snapshot] Should handle deleteScreenshotStorages', () => {
    const action: ActionType<typeof actions> = {
      type: getType(actions.deleteScreenshotStorages),
    }

    const currentState: IScreenShotStorageState = {
      availableStorage: null,
      status: true,
    }

    expect(screenshots(currentState, action)).toMatchSnapshot()
  })
})
