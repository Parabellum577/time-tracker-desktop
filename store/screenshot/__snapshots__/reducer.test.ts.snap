// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Store] Screenshot Reducer [Failure] Should handle getScreenshotStorage 1`] = `
Object {
  "availableStorage": null,
  "status": true,
}
`;

exports[`[Store] Screenshot Reducer [Success] [Snapshot] Should handle addScreenshotStorage 1`] = `
Object {
  "availableStorage": Object {
    "Email": "user@gmail.com",
    "FolderName": "New Folder",
    "Image": "",
    "Name": "bla-bla-bla",
    "StorageType": "DROPBOX",
  },
  "availableStorages": Object {
    "Email": "user@gmail.com",
    "FolderName": "New Folder",
    "Image": "",
    "Name": "bla-bla-bla",
    "StorageType": "DROPBOX",
  },
  "status": true,
}
`;

exports[`[Store] Screenshot Reducer [Success] [Snapshot] Should handle deleteScreenshotStorages 1`] = `
Object {
  "availableStorage": null,
  "status": false,
}
`;

exports[`[Store] Screenshot Reducer [Success] [Snapshot] Should handle getScreenshotStorage 1`] = `
Object {
  "availableStorage": Object {
    "Email": "user@gmail.com",
    "FolderName": "New Folder",
    "Image": "",
    "Name": "bla-bla-bla",
    "StorageType": "DROPBOX",
  },
  "status": true,
}
`;
